src/wp-includes/script-loader.php

```diff
@@ -1843,6 +1843,32 @@ function wp_just_in_time_script_localization() {
 	);
 }
 
+
+/**
+ * Prints a loader script if there is text/template registered script.
+ *
+ * When added to the DOM, this script converts any text/template scripts
+ * associated with a handle into type/javascript, and executes them.
+ */
+function wp_print_template_loader_script() {
+	$wp_scripts = wp_scripts();
+
+	if ( $wp_scripts->has_delayed_inline_script() ) {
+		$output    = <<<JS
```

Comment:
Instead of inlining the JS here, how about pulling in the JS file as is done in `_print_emoji_detection_script()`? This will have the benefit of ensuring that linting tools will be able to check the JS for issues and contributors will get syntax-highlighting in their IDEs, for example.

https://github.com/10up/wordpress-develop/blob/b98b5e59d72a68bc31f99b866fa26f7769d16e97/src/wp-includes/formatting.php#L5856-L5859

Reply:
I like the DX enhancements that would come from loading this from a file, but worry a bit about the performance of adding an additional file read operation for what is a pretty small function compared to the size of the JS file being loaded for `wp-emoji-loader`.

If we wanted to do something like this, it might be better to build this function into a file that is already going to be enqueued instead.

Either way, I think this conversation is best had once the PR is open against core, so this change doesn't block us from getting additional community feedback.

Reply:
Note that there used to be two instances in core where scripts would be inlined in this way at build time via `grunt-include`: `_print_emoji_detection_script()` and `WP_Embed::get_post_embed_html()`. These were removed (by me) in 02a2f9c9f515eee1a10c4fa04d95b89c1a750caf (https://github.com/WordPress/wordpress-develop/pull/1745).